lib/phoenix/tracker/state.ex

```diff
@@ -478,15 +541,17 @@ defmodule Phoenix.Tracker.State do
     tag = tag(state)
     true = :ets.insert(state.values, {{topic, pid, key}, meta, tag})
     true = :ets.insert(state.pids, {pid, topic, key})
+    true = :ets.insert(state.tags, {tag, {topic, pid, key}})
     new_delta = %State{delta | values: Map.put(delta.values, tag, {pid, topic, key, meta})}
     %State{state | delta: new_delta}
   end
 
   @spec remove(t, pid, topic, key) :: t
-  defp remove(%State{pids: pids, values: values} = state, pid, topic, key) do
+  defp remove(%State{pids: pids, tags: tags, values: values} = state, pid, topic, key) do
     [{{^topic, ^pid, ^key}, _meta, tag}] = :ets.lookup(values, {topic, pid, key})
-    1 = :ets.select_delete(values, [{{{topic, pid, key}, :_, :_}, [], [true]}])
-    1 = :ets.select_delete(pids, [{{pid, topic, key}, [], [true]}])
+    :ets.delete(values, {topic, pid, key})
```

Comment:
what's the reasoning behind these changes?

Reply:
https://github.com/salemove/phoenix_pubsub/compare/a27ea2e71f2a7b154e178d6a82bf3f93b7536586..e22cc849a857c598d088c5e86fdab6def6f07425 now using the same function as with observe_removes

but explanation for previous change: select_delete uses match spec to do the delete. however for values and tags we already now the key. So we don't need to match and can just use regular delete. for pids I just changed select_delete to match_delete which is a bit shorter and because we don't need to know how many rows were actually deleted.